Psyche/Assets/Scripts/Player/ToolManagers/ToolManager.cs

```diff
@@ -1,89 +1,115 @@
+    public int MaxLevel;
+    public bool ToolEnabled;
 
-    //Tool State
-    public int level;
-    public Dictionary<int, Dictionary<InventoryManager.Element, ushort>> levelRequirements;
-    public int maxLevel;
 
-    public bool toolEnabled;
+    //================================================== Events ==================================================
+    public event Action<ToolDirective, bool, string, Dictionary<InventoryManager.Element, ushort>, int, int> ToolManagerUpdate;
 
+    //=================================================== Enum ===================================================
+    /// <summary>
+    /// Directive for actions on the <see cref="InventoryManager.Tool"/>
+    /// </summary>
     public enum ToolDirective
     {
-        Upgrade, 
+        /// <summary>
+        /// Upgrading the <see cref="InventoryManager.Tool"/>
+        /// </summary>
+        Upgrade,
+
+        /// <summary>
+        /// Provide info about the <see cref="InventoryManager.Tool"/>
+        /// </summary>
         Info,
+
+        /// <summary>
+        /// Maximum level of the <see cref="InventoryManager.Tool"/>
+        /// </summary>
         MaxLevel
     }
 
-    // Events
-    public event Action<ToolDirective, bool, string, Dictionary<InventoryManager.Element, ushort>, int, int> ToolManagerUpdate;
-
+    //=============================================== Tool Actions ===============================================
     /// <summary>
-    /// Enables the tool
+    /// Enable the <see cref="InventoryManager.Tool"/>
     /// </summary>
     public void Enable()
     {
-        toolEnabled = true;
-        ToolManagerUpdate?.Invoke(ToolDirective.Info, false, toolName, levelRequirements[level + 1], level + 1, maxLevel);
+        ToolEnabled = true;
+        ToolManagerUpdate?.Invoke(ToolDirective.Info, false, ToolName, LevelRequirements[Level + 1], Level + 1, MaxLevel);
     }
 
     /// <summary>
-    /// Used for tool activation
+    /// Used for <see cref="InventoryManager.Tool"/> activation
     /// </summary>
     public abstract void Activate();
 
     /// <summary>
-    /// Class-specific implementation to update local variables
+    /// Upgrades the <see cref="InventoryManager.Tool"/>
     /// </summary>
     protected abstract void UpgradeTool();
 
     /// <summary>
-    /// Modifies the tool when called
+    /// Modifies the <see cref="InventoryManager.Tool"/> when called
     /// </summary>
     public void Modify()
     {
         // Check for upgrading
-        int nextLevel = (levelRequirements.ContainsKey(level + 1) ? level + 1 : level);
+        int nextLevel = (LevelRequirements.ContainsKey(Level + 1) ? Level + 1 : Level);
         bool upgraded = true;
-        if (levelRequirements.ContainsKey(level + 1))
+        if (LevelRequirements.ContainsKey(Level + 1))
         {
-            foreach (var requirement in levelRequirements[level + 1])
+            foreach (var requirement in LevelRequirements[Level + 1])

```

Comment:
Should the var from this foreach be removed? If not then I don't see any other errors.

Reply:
Fixed